### PR TITLE
Clean up old settings code related to the grouped settings flag.

### LINKED ISSUE #2050: Clean up old settings code
## Description

Clean up the old settings code, once the "grouped settings" flag has been removed.

The `WC_Payments_Features::is_grouped_settings_enabled()` should be considered to always return `true`. And the `_wcpay_feature_grouped_settings` flag should not be used anywhere.

- Remove all code referencing the `_wcpay_feature_grouped_settings` flag
- Remove all code referencing the `WC_Payments_Features::is_grouped_settings_enabled()` flag (including all the frontend code for Express Payments that uses legacy options)
- Clean up the WCPay Dev tools

### PR DESCRIPTION
**Pull request by brettshumaker**
Fixes #2050 

#### Changes proposed in this Pull Request

This PR removes references to the `_wcpay_feature_grouped_settings` option as well as `WC_Payments_Features::is_grouped_settings_enabled()`. Refactors some code where necessary, and updated/removed tests that related to grouped settings. It also removes some front end code referencing custom and branded buttons that utilize old settings options.

I think I've removed everything I needed to and nothing that I didn't but would appreciate a thorough review on this.

#### Testing instructions

Check to make sure that all "Grouped Settings" features are working properly.

-------------------

- [ ] Added changelog entry (or does not apply)
- [ ] Covered with tests (or have a good reason not to test in description ☝️)

**Post merge**

- [ ] Added testing instructions to the [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) (or does not apply)

### CODE REVIEW

File: client/payment-request/index.js [hunk 1]
```diff
@@ -311,34 +307,6 @@ jQuery( ( $ ) => {
...
-					// We fallback to default paymentRequest button if no custom button is found in the UI.
-					// Add flag to be sure that created button is custom button rather than fallback element.
-					button.data( 'isCustom', true );
-					return button;
-				}
-			}
-
-			if ( wcpayPaymentRequestParams.button.is_branded ) {
-				if ( shouldUseGooglePayBrand() ) {
-					button = wcpayPaymentRequest.createGooglePayButton();
```
Comment (frosso) [inline client/payment-request/index.js:327]: The `createGooglePayButton` method doesn't seem to be used anywhere else anymore - can you please remove it?

File: client/payment-request/blocks/payment-request-express.js [hunk 2]
```diff
@@ -7,9 +7,7 @@ import { Elements, PaymentRequestButtonElement } from '@stripe/react-stripe-js';
  * Internal dependencies
  */
 import { useInitialization } from './use-initialization';
-import { GooglePayButton } from './branded-buttons';
```
Comment (frosso) [inline client/payment-request/blocks/payment-request-express.js:10]: The `GooglePayButton` component and its styles (`.gpay-button`) don't seem to be used anywhere else - can you please remove them?

File: client/payment-request/blocks/payment-request-express.js [hunk 3]
```diff
@@ -7,9 +7,7 @@ import { Elements, PaymentRequestButtonElement } from '@stripe/react-stripe-js';
...
-import { CustomButton } from './custom-button';
```
Comment (frosso) [inline client/payment-request/blocks/payment-request-express.js:11]: The `CustomButton` component and its styles (`#wcpay-custom-button`) don't seem to be used anywhere else - can you please remove them?

File: includes/class-wc-payments-payment-request-button-handler.php [hunk 4]
```diff
@@ -1478,36 +1439,19 @@ public function get_option_is_apple_pay_enabled( $value ) {
...
-			'is_custom'    => $this->is_custom_button(),
-			'is_branded'   => $this->is_branded_button(),
-			'css_selector' => $this->custom_button_selector(),
-			'branded_type' => $this->gateway->get_option( 'payment_request_button_branded_type' ),
+			'branded_type' => 'default' === $button_type ? 'short' : 'long',
+			// these values are no longer applicable.
+			'css_selector' => '',
+			'label'        => '',
+			'is_custom'    => false,
+			'is_branded'   => false,
```
Comment (frosso) [inline includes/class-wc-payments-payment-request-button-handler.php:1454]: We can remove these too

File: tests/unit/test-class-wc-payment-gateway-wcpay.php [hunk 5]
```diff
@@ -112,8 +112,6 @@ public function tearDown() {
 		// Fall back to an US store.
 		update_option( 'woocommerce_store_postcode', '94110' );
 		$this->wcpay_gateway->update_option( 'saved_cards', 'yes' );
-
-		delete_option( '_wcpay_feature_grouped_settings' );
```
Comment (frosso) [inline tests/unit/test-class-wc-payment-gateway-wcpay.php:116]: There's another test case referencing the `_wcpay_feature_grouped_settings` flag, can the `update_option( '_wcpay_feature_grouped_settings', '1' );` be removed as well?

**Review by frosso — COMMENTED** [review]
Great job with the cleanup! It looks like there are a few remaining pieces.
Including this: https://github.com/Automattic/woocommerce-payments/blob/487284e6f4bce31fb3382fa8aba816df841a3477/client/settings/index.js#L46-L68 (the selected ones are the only used elements, the rest can be removed)

I tested this on JurassicNinja with both Chrome and Safari:
Screen Shot 2021-08-20 at 9 21 11 AM
Screen Shot 2021-08-20 at 9 19 13 AM

**Comment by brettshumaker** [comment]
@frosso Thanks for double-checking to see if I had gotten everything - clearly I hadn't! I've addressed everything you mentioned above in this most recent commit, so I think I'm good now.

File: client/settings/index.js [hunk 6]
```diff
@@ -43,18 +41,6 @@ window.addEventListener( 'load', () => {
...
-);
-if ( settingsContainer ) {
-	ReactDOM.render(
-		<WCPaySettingsContext.Provider value={ wcpaySettings }>
-			<SettingsManager />
-		</WCPaySettingsContext.Provider>,
-		settingsContainer
-	);
-}
-
```
Comment (frosso) [inline client/settings/index.js:57]: Sorry for the confusion in my comment - this is needed.
The parts that are not needed are
https://github.com/Automattic/woocommerce-payments/blob/487284e6f4bce31fb3382fa8aba816df841a3477/client/settings/index.js#L16-L40
https://github.com/Automattic/woocommerce-payments/blob/487284e6f4bce31fb3382fa8aba816df841a3477/client/settings/index.js#L70-L181

Comment (brettshumaker) [reply client/settings/index.js]: Whoops! My mistake. I've added that bit back.

Comment (frosso) [reply client/settings/index.js]: I'm still seeing the settings code that is unrelated to the newer settings.
You'll need to remove L16-40 and L70-181 from `woocommerce-payments/client/settings/index.js`

Comment (brettshumaker) [reply client/settings/index.js]: > L16-40 and L70-181 from woocommerce-payments/client/settings/index.js
Yikes. I've really misunderstood what you were referencing, then! I've removed those lines as well as line 6 as `__` was no longer used.

**Review by frosso — APPROVED** [review]
Gave it another run, looks good! Thanks for the changes!

**Pull request merged**